### PR TITLE
Connection readiness of k8s client gets ns

### PR DESCRIPTION
**Pull request by nebril**
kube-system is always in the cluster, getting it indicates that
connection to apiserver is ready.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8855)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Review by tgraf — CHANGES_REQUESTED** [review]
Please also remove the "componentstatus" from the ClusterRole so we have less expansive RBAC requirements.

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by ianvernon** [comment]
Build failure:

Appears related to this change: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/14439/

```
019-08-16T18:52:36.224298288Z level=error msg="Unable to contact k8s api-server" error="namespaces \"kube-system\" is forbidden: User \"system:serviceaccount:kube-system:cilium-operator\" cannot get resource \"namespaces\" in API group \"\" in the namespace \"kube-system\"" ipAddr="https://10.96.0.1:443" subsys=k8s
2019-08-16T18:52:36.224312617Z level=fatal msg="Unable to connect to Kubernetes apiserver" error="unable to create k8s client: unable to create k8s client: namespaces \"kube-system\" is forbidden: User \"system:serviceaccount:kube-system:cilium-operator\" cannot get resource \"namespaces\" in API group \"\" in the namespace \"kube-system\"" subsys=cilium-operator
```

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

File: pkg/k8s/client.go [hunk 1]
```diff
@@ -128,9 +128,9 @@ func CreateClient(config *rest.Config) (*kubernetes.Clientset, error) {
 	return cs, err
 }
 
-// isConnReady returns the err for the controller-manager status
+// isConnReady returns the err for the kube-system namespace get
 func isConnReady(c *kubernetes.Clientset) error {
-	_, err := c.CoreV1().ComponentStatuses().Get("controller-manager", metav1.GetOptions{})
```
Comment (vsxen) [inline pkg/k8s/client.go:133]: cilium was crashloopbackoff .
```
level=error msg="Unable to contact k8s api-server" error="componentstatuses \"controller-manager\" is forbidden: User \"system:serviceaccount:kube-system:cilium\" cannot get resource \"componentstatuses\" in API group \"\" at the cluster scope" ipAddr="https://10.96.0.1:443" subsys=k8s
level=fatal msg="Unable to initialize Kubernetes subsystem" error="unable to create k8s client: unable to create k8s client: componentstatuses \"controller-manager\" is forbidden: User \"system:serviceaccount:kube-system:cilium\" cannot get resource \"componentstatuses\" in API group \"\" at the cluster scope" subsys=daemon
```
with this docker image
`cilium/cilium                                                                  v1.6.0              7bd190b404f4        6 days ago          710MB`

you should change this yaml file URL 

https://cilium.readthedocs.io/en/v1.6/gettingstarted/minikube/#install-cilium

Comment (brb) [reply pkg/k8s/client.go:133]: @vsxen Thanks for reporting. This should be fixed after https://github.com/cilium/cilium/pull/9038 has been merged.

**Pull request merged**